share/eds/eds_test.go

```diff
@@ -130,6 +131,28 @@ func TestInnerNodeBatchSize(t *testing.T) {
 	}
 }
 
+func TestReadEDS(t *testing.T) {
+	eds := writeRandomEDS(t)
+	dah := da.NewDataAvailabilityHeader(eds)
+	f := openWrittenEDS(t)
+	defer f.Close()
+
+	loaded, err := ReadEDS(context.Background(), f, dah)
+	require.NoError(t, err, "error reading EDS from file")
+	require.Equal(t, eds.RowRoots(), loaded.RowRoots())
+	require.Equal(t, eds.ColRoots(), loaded.ColRoots())
+}
+
+func TestReadEDSContentIntegrityMismatch(t *testing.T) {
+	writeRandomEDS(t)
+	dah := da.NewDataAvailabilityHeader(share.RandEDS(t, 4))
+	f := openWrittenEDS(t)
+	defer f.Close()
+
+	_, err := ReadEDS(context.Background(), f, dah)
+	require.Error(t, err)
```

Comment:
maybe check that the error is the content integrity err

Reply:
done